be/src/util/mem_info.cpp

```diff
@@ -117,4 +112,47 @@ std::string MemInfo::debug_string() {
     return stream.str();
 }
 
+void MemInfo::set_memlimit_if_container() {
+    // check if application is in docker container or not via /.dockerenv file
+    bool running_in_docker = fs::path_exist("/.dockerenv");
+    if (running_in_docker) {
+        struct statfs fs;
+        int err = statfs("/sys/fs/cgroup", &fs);
+        if (err < 0) {
+            LOG(WARNING) << "Fail to get file system statistics. err: " << errno_to_string(err);
+            return;
+        }
+
+        std::ifstream memoryLimit;
+        std::string line;
+        StringParser::ParseResult result = StringParser::PARSE_SUCCESS;
+        if (fs.f_type == TMPFS_MAGIC) {
+            // cgroup v1
+            // Read from /sys/fs/cgroup/memory/memory.limit_in_bytes
+            memoryLimit.open("/sys/fs/cgroup/memory/memory.limit_in_bytes");
+            getline(memoryLimit, line);
+            _s_physical_mem = StringParser::string_to_int<int64_t>(line.data(), line.size(), &result);
+        } else if (fs.f_type == CGROUP2_SUPER_MAGIC) {
+            // cgroup v2
+            // Read from /sys/fs/cgroup/memory/memory.max
+            memoryLimit.open("/sys/fs/cgroup/memory.max");
+            getline(memoryLimit, line);
+
+            if (line == "max") {
+                _s_physical_mem = std::numeric_limits<int64_t>::max();
```

Comment:
how will starrocks use `_s_physical_mem`?

I feel like you shall to use `memory.min` and `memory.low` which guarantee the minimal amount of memory that a BE pod can get instead of the maximum amount of memory that it can get.
K8S use `memory.min` and `memory.low` for memory resource request.
check it out here:
https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#how-pods-with-resource-limits-are-run

![image](https://user-images.githubusercontent.com/5108773/206359026-ba0bb393-8054-4a11-bd04-f6fb0fc67ace.png)